lib/chess/core/cell.rb

```diff
@@ -0,0 +1,82 @@
+require 'colorize'
+require 'colorized_string'
+
+require_relative './chess'
+
+module Chess
+  module Core
+    class Cell
+      attr_reader :algebraic, :coordinates
+
+      Algebraic = Struct.new(:column, :row) do
+        def to_s
+          "#{column}#{row}"
+        end
+      end
+
+      Coordinates = Struct.new(:column, :row) do
+        def to_a
+          [row, column]
+        end
+      end
+
+      def initialize(name, fill_color, occupied = false, occupied_by = nil)
+        @occupied = occupied
+        @occupied_by = occupied_by
+        @fill_color = fill_color
+        @name = name
+
+        set_nomenclature_values
+      end
+
+      def occupy_with(piece)
+        @occupied = true
+        @occupied_by = piece
+      end
+
+      def free
+        @occupied = false
+        @occupied_by = nil
+      end
+
+      def occupant
+        @occupied_by
```

Comment:
you could rename this attribute to `occupant` and add a `attr_reader` to replace this method

Reply:
done